assets/app/scripts/directives/oscFileInput.js

```diff
@@ -8,15 +8,79 @@ angular.module('openshiftConsole')
         model: "=",
         required: "=",
         disabled: "=ngDisabled",
-        helpText: "@?"
+        helpText: "@?",
+        dropArea: "@?"
       },
       templateUrl: 'views/directives/osc-file-input.html',
       link: function(scope, element){
         scope.helpID = _.uniqueId('help-');
         scope.supportsFileUpload = (window.File && window.FileReader && window.FileList && window.Blob);
         scope.uploadError = false;
+
+        var dropArea = scope.dropArea || element,
+        highlightDropArea = false,
+        showDropArea = false;
+
+        scope.$watch('disabled', function() {
+          if (scope.disabled) {
+            // Remove listeners for page drag & drop actions
+            $(dropArea).off('dragover');
+            $(dropArea).off('dragleave');
+            $(dropArea).off('drop');
+          } else {
+            // Add listeners for page drag & drop actions
+            $(dropArea).on('dragover', function() {
+              $(dropArea).addClass('highlight-drag-and-drop-area');
+              highlightDropArea = true;
+            });
+            $(dropArea).on('dragleave', function() {
+              highlightDropArea = false;
+              _.delay(function(){
+                  if( !highlightDropArea ){ $(dropArea).removeClass('highlight-drag-and-drop-area'); }
+              }, 200 );
+            });
+            $(dropArea).on('drop', function(e) {
+              preventAndStop(e);
+              var files = _.get(e, 'originalEvent.dataTransfer.files', []);
+              if (files.length > 0 ) {
+                scope.file = _.head(files);
+                $(element).trigger('change');
+              }
+              removeDropAreaClasses();
+              return false;
+            });
+          }
+        }, true);
+
+        $(document).on('dragover', function(e) {
+          preventAndStop(e);
+          if (!scope.disabled) {
+            $(dropArea).addClass('show-drag-and-drop-area');
+            showDropArea = true;
+          }
+        });
+        $(document).on('dragleave', function(e) {
+          preventAndStop(e);
+          if (!scope.disabled) {
+            showDropArea = false;
+            _.delay(function(){
+                if( !showDropArea ){ $(dropArea).removeClass('show-drag-and-drop-area'); }
+            }, 200 );
+          }
+        });
+        $(document).on('drop', function(e) {
+          preventAndStop(e);
+          removeDropAreaClasses();
+          return false;
+        });
```

Comment:
Had to leave the registration of these three listeners outside of the `scope.$watch('disabled')` since we need then to be registered all the time.